compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -4344,10 +4338,15 @@ private BType getStructuredBindingPatternType(BLangVariable bindingPatternVariab
                         ((BMapType) ((BLangSimpleVariable) recordVariable.restParam).type).constraint :
                         symTable.anydataType;
             }
+            BLangRecordTypeNode recordTypeNode = createRecordTypeNode(typeDefFields, recordVarType);
```

Comment:
This was done to fix NPE in BirGen in match statement due to scope entries and init function misssing when creating record type.